kettle/make_json.py

```diff
@@ -188,7 +188,18 @@ def parse_result(child_node):
         time = float(child_node.attrib.get('time') or 0) #time val can be ''
         failure_text = None
         for param in child_node.findall('failure'):
-            failure_text = param.text or param.attrib.get('message', 'No Failure Message Found')
+            msg_attribute = param.attrib.get('message')
+            msg_data = param.text
+            if msg_attribute and msg_data:
+                # Use both. With Ginkgo V2, the attribute contains the failure message
+                # and the data contains the stack backtrace. We could leave out the
+                # stack backtrace, but the function names in it might be useful
+                # to distinguish different unrelated failures that happen to have the
+                # same message.
+                failure_text = msg_attribute + "\n" + msg_data
```

Comment:
Longer failure text has performance implications. It takes longer to cluster up, and we've had issues before where it generated too much data for go.k8s.io/triage to effectively load on a browser (ref: https://github.com/kubernetes/test-infra/issues/22906). As I type this I notice triage clustering time is on the rise again (https://testgrid.k8s.io/sig-testing-misc#triage&graph-metrics=test-duration-minutes&width=5)

I have a suspicion that stack traces are effectively noise, as they're going to diverge more and more the lower they get.

If we go this approach I would suggest putting in well-known delimiters that make it easier for triage to parse out "message" vs  "data".  A newline isn't going to be enough, since messages already have newlines in them.

Changes to triage's clustering algorithm will take effect much more quickly.  Changes to kettle are going to take O(days) to really start showing up in triage.

Reply:
We don't have a convention for failure messages that would allow us to separate "message" from "additional values".
If we want to keep the overall failure message small, how about drastically lowering the `maxFailureMessageSize` limit in https://github.com/kubernetes/kubernetes/pull/113384?

> stack traces are effectively noise, as they're going to diverge more and more the lower they get.

They should always start at the actual test code and then don't go much higher because Ginkgo v2 runs tests in fresh goroutines. I don't think we record very deep backtraces at the moment.

Reply:
Even better, Ginko also truncates above the test code:
```
k8s.io/kubernetes/test/e2e/node.createAndTestPodRepeatedly(0x3, 0xf, {0x7f6b4b8?, 0xc000d60aa0}, {0x7f8b7c0?, 0xc000d609e0})
	test/e2e/node/pods.go:474 +0x81d
k8s.io/kubernetes/test/e2e/node.glob..func10.3.3()
	test/e2e/node/pods.go:218 +0xaf}
```